demo/ARContainerViewController.swift

```diff
@@ -92,9 +89,29 @@ final class ARContainerViewController: UIViewController {
 }
 
 extension ARContainerViewController: NavigationServiceDelegate {
-    
+
     func navigationService(_ service: NavigationService, didUpdate progress: RouteProgress, with location: CLLocation, rawLocation: CLLocation) {
         instructionsLabel.isHidden = false
         instructionsLabel.text = progress.currentLegProgress.currentStep.instructions
+
+        if routeHasChanged {
+            routeHasChanged = false
+            navigationDelegate?.navigationService(didUpdate: Route(route: progress.route))
+        }
+
+        if progress.currentLegProgress.userHasArrivedAtWaypoint {
+            routeHasChanged = true
+            navigationDelegate?.navigationServiceArrivedAtDestination()
+        }
+    }
+
+    func navigationService(_ service: NavigationService, didRerouteAlong route: MapboxDirections.Route, at location: CLLocation?, proactive: Bool) {
+        routeHasChanged = true
     }
 }
+
+
+public protocol NavigationManagerDelegate: class {
+    func navigationService(didUpdate route: MapboxVisionARNative.Route)
```

Comment:
I suggest to start a method's name with the name of type is delegating (e.g. `navigationManagerDid...` instead of `navigationServiceDid...`).

Reply:
Especially we have a separate `NavigationService` type.

Reply:
Renamed delegate